Library/Formula/qt.rb

```diff
@@ -25,6 +25,9 @@ class Qt < Formula
     sha256 "c8a0fa819c8012a7cb70e902abb7133fc05235881ce230235d93719c47650c4e"
   end
 
+  # Make `HOMEBREW_PREFIX/lib/qt4/plugins` an additional plugin search path.
+  patch :DATA
```

Comment:
> What were the alternatives you considered?

I tried to give an idea in my initial post (see the bullet list). Haven't fully articulated them there, though, so could do this if you'd like to hear the details.